app/scripts/components/common/map/style-generators/hooks.ts

```diff
@@ -206,6 +225,59 @@ export function useWMS({ id, stacCol, stacApiEndpointToUse, onStatusChange }) {
   };
 }
 
+export function useWMTS({ id, stacCol, stacApiEndpointToUse, onStatusChange }) {
+  const defaultBounds: [number, number, number, number] = [-180, -90, 180, 90];
+  const [wmtsUrl, setWmtsUrl] = useState<string[]>();
+  const [bounds, setBounds] =
+    useState<[number, number, number, number]>(defaultBounds);
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    async function load() {
+      try {
+        onStatusChange?.({ status: S_LOADING, id });
+        const data: WMTSResponseData = await requestQuickCache({
+          url: `${stacApiEndpointToUse}/collections/${stacCol}`,
+          method: 'GET',
+          controller
+        });
+        const bounds = data.extent.spatial.bbox[0];
+        setBounds(bounds);
+        const wmtsLink = data.links.find((l) => l.rel === 'wmts');
+
+        if (!wmtsLink || !wmtsLink.href) {
+          throw new Error('WMTS link is missing or malformed');
+        }
+        const primaryHref = wmtsLink.href;
+        const alternateServers: string[] = wmtsLink['href:servers'] ?? [];
```

Comment:
If I am reading the code right,  `useWMS` and `useWMTS` hooks appear to differ only in how they handle `alternateServers`. This makes me think we could consolidate them into a single hook.
Here's the expected behavior:

WMS layers: Since wmsLink lacks href:servers, it would return an empty array [], causing the hook to return only [primaryHref] (We will need to adjust wms-timeseries to accept wmsurl as string[])
WMTS layers: These would return both [primaryHref, alternateServers] since the alternate servers exist